src/containers/blocks/UserBlock.js

```diff
@@ -0,0 +1,127 @@
+import {handleDragOver, handleDrop} from "../../functions/handleDragAndDrop";
+import {addNewFilter, toggleOptionsBox} from "../../actions";
+import DropContainer from "./DropBlock"
+
+class UserBlockContainer extends Component {
+
+  handleFilterDrop(ev, parentItem) {
+    const droppedItem = handleDrop(ev);
+    if(droppedItem) {
+      this.props.addNewFilter(ev, droppedItem, parentItem);
+    }
+  }
+
+  toggleOptionsBox(ev, test, ID, parentItem) {
+    ev.stopPropagation();
+    this.props.toggleOptionsBox(test, ID, parentItem);
+  }
+
+  generateListOfUserBlocks(type) {
+    if(this.props.test.length > 0) {
+      return this.props.test.map((test, index) => {
+        if(typeof test !== 'undefined') {   
+          if(test.dropTo.toLowerCase() === type) {
+            const elemClass = type + "-list-container";
+            let elemID = null;
+            if(test.type === "Source W3CHTML5") {
+              elemID = "source-comparators-" + index;
+            } else {
+              elemID = generateID(test) + "-" + index;
+            }
+            if(type === "comparators") {
+              return (
+              <div key={elemID} >
+                <div className={elemClass}>
+                  <div 
+                  className="block custom nested" 
+                  id={elemID} 
+                  onClick={(ev) => this.toggleOptionsBox(ev, test, elemID)}
+                  onDragOver={(ev) => handleDragOver(ev)}  
+                  onDrop={(ev) => this.handleFilterDrop(ev, test)}
+                  >{test.type}</div>
+                  {this.generateListOfFilters(test, index)}
+                </div>
+                <DropContainer dropTo={"comparators"} />
+              </div>
+              )
+            } else if(type === "modifiers") {
+              return (
+                <div className={elemClass}  key={elemID}>
+                  <div 
+                  className="block custom nested" 
+                  onClick={(ev) => this.toggleOptionsBox(ev, test, elemID)}
+                  key={elemID} 
+                  id={elemID}>
+                  <span>{test.type}</span>
+                  </div>
+                  <DropContainer dropTo={"modifiers"} />
+                </div>
+              )
+            } else {
+              return (
+                <div className={elemClass}  key={elemID}>
+                  <div 
+                  className="block custom nested" 
+                  onClick={(ev) => this.toggleOptionsBox(ev, test, elemID)}
+                  key={elemID} 
+                  id={elemID}>
+                  <span>{test.type}</span>
+                  </div>
+                  <DropContainer dropTo={"collectors"} />
+                </div>
+              )
+            }
+          } else {
+            return null;
+          }
+        }
+        return null;
+      });
+    }
+  }
+
+  generateListOfFilters(test, parentIndex) {
+    if(test.filters !== null) {
```

Comment:
You can use shorthand condition for snippets like this (test.filters !== null && this.filters.map ... )

Reply:
Didn't know that one, done.